sync-android/src/main/java/com/cloudant/sync/sqlite/android/encryption/AttachmentEncryptionUtils.java

```diff
@@ -0,0 +1,274 @@
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Created by estebanmlaver on 5/13/15.
+ */
+public class AttachmentEncryptionUtils {
+
+    private static final String UTF_8 = "UTF-8"; //$NON-NLS-1$
+    private static final int PBKDF2_LENGTH_BYTE_SIZE = 32;
+    private static final int IV_NUM_BYTES = 16;
+    private static final int BYTES_FOR_SALT = 32;
+
+    public static final String CYPHER_TEXT_LABEL = "ct"; //$NON-NLS-1$
+    public static final String IV_LABEL = "iv"; //$NON-NLS-1$
+    public static final String ENCRYPTION_SOURCE_LABEL = "src"; //$NON-NLS-1$
+    public static final String VERSION_LABEL = "v"; //$NON-NLS-1$
+
+    private static final String ENCRYPTION_SOURCE = "java"; //$NON-NLS-1$
+
+    private static com.cloudant.sync.sqlite.android.encryption.AttachmentEncryptionUtils singleton;
+
+    private AttachmentEncryptionUtils(){ }
+
+    /**
+     * Only return one instance of this factory
+     * @return
+     */
+    public static synchronized AttachmentEncryptionUtils getInstance() {
+        if (singleton == null) {
+            singleton = new AttachmentEncryptionUtils();
+        }
+        return singleton;
+    }
+
+    public Cipher getCipher(String key, File attachFile, int mode, byte[] iv) throws SecurityUtilsException {
+        if(key == null || key.length() <= 0){
+            throw new SecurityUtilsException("Key cannot be null or empty.");
+        }
+
+        if(attachFile == null || attachFile.length() <= 0){
+            throw new SecurityUtilsException("Attachement cannot be null or empty.");
+        }
+
+        Cipher cipher = null;
+        try {
+            //Args for mode (encrypt or decrypt), sqlcipher key in bytes, and IV in bytes
+            cipher = initCipher(mode, hexStringToByteArray(key), iv);
+
+            //byte[] inputBytes = new byte[(int) attachFile.length()];
+            //byte[] outputBytes = cipher.doFinal(inputBytes);
+        } catch (InvalidKeyException e1) {
+            throw new SecurityUtilsException("Problem occured while encrypting. Make sure the given key is valid.", e1);
+        } catch (NoSuchAlgorithmException e1) {
+            throw new SecurityUtilsException("Problem occured while encrypting. Make sure the given key is valid.", e1);
+        } catch (NoSuchPaddingException e1) {
+            throw new SecurityUtilsException("Problem occured while encrypting. Make sure the given key is valid.", e1);
+        } catch (InvalidAlgorithmParameterException e1) {
+            throw new SecurityUtilsException("Problem occured while encrypting. Make sure the given key is valid.", e1);
+        }
+
+        return cipher;
+    }
+
+    /**
+     * Encrypt the given text with the given key. The encrypted text is included in the JSONObject that is returned.
```

Comment:
Yea, this comment was copied and needs to be completely changed. The method changed a lot since it's initial state.